### DISCUSSION TITLE
[发现错误?] GitHub 搜索框光标错位

### DISCUSSION THREAD

**Discussion by cxw620** [F9y4ng/GreasyFork-Scripts#376] [问答专区(Question / Answer)]
## 1. 准备工作（确认预先完成的操作）

- [x] 我确认正在使用的是当前脚本的最新版本。
- [x] 我已在Issues或Discussions中搜索过此问题，但没有找到相关答案。
- [x] 我已在关闭浏览器其他扩展、仅开启此脚本的情况下测试，但依然出错。

## 2. 描述错误信息（清楚而简洁地描述错误）

如题

## 3. 错误过程再现（请按示例填写行为步骤，直至复现您遇到的问题）

如题

## 4. 屏幕截图（如有屏幕截图，请添加以帮助解释您的问题）

-

## 5. 预期结果（清晰明了地描述您所期待想要什么样的结果）

-

## 6. 客户端信息（请在“**冒号**”后认真填写您的本地信息）
- 操作系统：Windows 11 专业工作站版 24H2 26100.1742 Windows Feature Experience Pack 1000.26100.18.0
- 浏览器：Firefox
- 浏览器版本：131.0 (20240923135042)
- 脚本管理器：Tampermonkey
- 脚本管理器版本：v5.1.1
- 所用的脚本名称：字体渲染（自用脚本）
- 所用的脚本版本：2024.09.11.1
- 其他补充信息：

已完全重置插件

## 7. 脚本运行环境排查 （请确认环境条件并填写相关信息）
- 是否同时运行了其他浏览器扩展？

Dark Reader, 沉浸式翻译

已测试全部关闭其他插件.

- 是否同时运行了其他第三方用户脚本？

无

## 8. 附加信息（在此处添加您认为有用的其他信息）

F12 调试 CSS 排查得知此项导致了错误, 暂时禁用后恢复正常

![图片](https://github.com/user-attachments/assets/2f1b0d22-9792-4f77-b291-7c151de6fc66)

![图片](https://github.com/user-attachments/assets/4127608a-cce8-4456-b2b6-c570be2a5b1d)

![图片](https://github.com/user-attachments/assets/7369f5cf-e16e-4b82-8080-c41b2b575f86)

但是这个应该也是为了修别的问题加进来的...

**Comment by F9y4ng** [comment]
这个问题仅出现在Firefox浏览器下，造成的原因就是Firefox浏览器下的input的标签在未显式定义宽度时，会被不同字宽的字体撑开容器宽度，为了解决这个问题，只能将input标签的字体初始化为浏览器默认字体。而github的搜索框显示的是input内容，编辑的是div标签的内容，也就是光标位置随div内容改变，所以就出现了你提到的问题。

由于去除input标签字体的初始化后，在某些站点与字体热替换会出现冲突，从而造成input被撑爆的情况，所以，我暂未找到比较完美的解决方案。所以要么等Firefox在3000年之前修复这个Bug，要么等我发现新的解决途径。

**Reply by cxw620** [reply]
能否特殊处理一下 Github 的搜索框呢() 毕竟还是常用场景()

**Reply by F9y4ng** [reply]
我会在这个月着手处理这个问题，目前已知的绝大部分 Discuz 论坛默认样式都是存在此问题，是否还有其他具有普遍性的情况，还需要进一步探查。

如何启用/在哪启用 针对在 Firefox 中 input 标签的样式修复也要重新确定，毕竟参数设置界面上寸土寸金😕，但它仍然是一个极为小众的功能需求。

**Comment by F9y4ng** [comment]
请更新至 2024.11.02.1

相关在Firefox浏览器下的使用说明请查阅 [这里](https://github.com/F9y4ng/GreasyFork-Scripts/wiki/%E5%AD%97%E4%BD%93%E6%B8%B2%E6%9F%93%EF%BC%88%E8%87%AA%E7%94%A8%E8%84%9A%E6%9C%AC%EF%BC%89#firefox-input%E6%A0%B7%E5%BC%8F%E4%BF%AE%E5%A4%8D)，默认无需开启，遇到input标签样式异常时打开此开关。

如有其他普遍性的input标签样式问题，请提供测试地址，感谢。
